### PR TITLE
Polish the documentation for GA release

### PR DESCRIPTION
**Pull request by dzou**
Polish the documentation for GA release.

- Reworded/reorganized some things
- Merge table from cloud docs to our doc.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -156,21 +154,31 @@ The following sections below describe the feature set of the Cloud Spanner Hiber
 [options="header"]
 |===
 | Unsupported Feature | Description
-| Constraints | No support for `FOREIGN KEY`, `UNIQUE`, or `ON DELETE CASCADE`. Cloud Spanner does not enforce relationship links outside of Hibernate.
+| Constraint-based Annotations | The dialect does not support the database constraint-based annotations `@Column(unique = true)` and `@OnDelete(action = OnDeleteAction.CASCADE)`.
```
Comment (meltsufin) [inline README.adoc]:
```suggestion
| Schema Constraints  | Dialect schema generation does not generate foreign key and column uniqueness constraints for entity relationships and columns. Server-side cascading deletes are also not supported.
```

File: README.adoc [hunk 2]
```diff
@@ -156,21 +154,31 @@ The following sections below describe the feature set of the Cloud Spanner Hiber
...
 | Catalog and schema scoping for table names | Tables name references cannot contain periods or other punctuation.
-| Column default values | Cloud Spanner treats `null` like any other value and it gets no special handling.
+| Column default values | The dialect does not currently set default values based on the `@ColumnDefault` annotation, because Cloud Spanner does not support column defaults in the DDL.
 | Big-decimal or arbitrary-precision numbers | The dialect does not support `java.math.BigDecimal` because Cloud Spanner doesn't have SQL `NUMERIC` or `DECIMAL` support.
```
Comment (meltsufin) [inline README.adoc]: Can you add the row about Simba JDBC Driver not being supported?

**Comment by meltsufin** [comment]
Still need to copy the "Optimizing queries" section?

**Comment by dzou** [comment]
> Still need to copy the "Optimizing queries" section?

This is already in the User Guide section so we are good.

**Pull request merged**